src/shared/Microsoft.Git.CredentialManager/Application.cs

```diff
@@ -147,63 +146,41 @@ Task IConfigurableComponent.ConfigureAsync(ConfigurationTarget target)
-                    }
-                    break;
+            IGitConfiguration config = Context.Git.GetConfiguration(configLevel);
 
-                case ConfigurationTarget.System:
-                    // For machine-wide configuration, we are looking for the following to be set in the system config:
-                    //
-                    // [credential]
-                    //     helper = {appPath}
-                    //
-                    config = Context.Git.GetConfiguration(GitConfigurationLevel.System);
-                    string currentValue = config.GetValue(helperKey);
-                    if (Context.FileSystem.IsSamePath(currentValue, appPath))
-                    {
-                        Context.Trace.WriteLine("Credential helper system configuration is already set correctly.");
-                    }
-                    else
-                    {
-                        Context.Trace.WriteLine("Updating Git credential helper system configuration...");
-                        config.SetValue(helperKey, appPath);
-                    }
+            // We are looking for the following to be set in the config:
+            //
+            // [credential]
+            //     ...                # any number of helper entries (possibly none)
+            //     helper =           # an empty value to reset/clear any previous entries (if applicable)
+            //     helper = {appPath} # the expected executable value & directly following the empty value
+            //     ...                # any number of helper entries (possibly none)
+            //
+            string[] currentValues = config.GetRegex(helperKey, Constants.RegexPatterns.Any).ToArray();
 
-                    break;
+            // Try to locate an existing app entry with a blank reset/clear entry immediately preceding
+            int appIndex = Array.FindIndex(currentValues, x => Context.FileSystem.IsSamePath(x, appPath));
+            if (appIndex > 0 && string.IsNullOrWhiteSpace(currentValues[appIndex - 1]))
+            {
+                Context.Trace.WriteLine("Credential helper configuration is already set correctly.");
+            }
+            else
+            {
+                Context.Trace.WriteLine("Updating Git credential helper configuration...");
 
-                default:
-                    throw new ArgumentOutOfRangeException(nameof(target), target, "Unknown configuration target.");
+                // Clear any existing app entries in the configuration
+                config.UnsetAll(helperKey, Regex.Escape(appPath));
+
+                // Add an empty value for `credential.helper`, which has the effect of clearing any helper value
+                // from any lower-level Git configuration, then add a second value which is the actual executable path.
+                config.ReplaceAll(helperKey, Constants.RegexPatterns.None, string.Empty);
```

Comment:
Are we really sure this should be `ReplaceAll`? I thought maybe `Add` or `Append`...

Reply:
The manpages say:
```
--add
           Adds a new line to the option without altering any existing values. This is the same as providing
           ^$ as the value_regex in --replace-all.
```

Reply:
Will replace with calls directly to `-add` however for good measure.

Reply:
Done. Replaced these usages of `--replace-all` with a regex matching nothing, with `--add`.
Also updated our use of `--get-regexp` with and "any" regex value to `--get-all`.